src/averageAges.js

```diff
@@ -36,14 +47,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  const filteredPeople = people.filter(
+    ({ name, sex }) => sex === 'f'
+      && (!withChildren || people.some((person) => person.mother === name))
```

Comment:
Desctucturing is ok, but somewhere in the code you use it, somewhere you don't. I think it's better to stick to the same style.